### PR TITLE
#8976: show sidebar brick not showing active tab

### LINKED ISSUE #8976: Show Sidebar Brick not showing active tab
## Describe the bug

The correct sidebar panel is not always being displayed when calling the `Show Sidebar` brick in a mod.
The sidepanel is supposed to open:
  1) the panel that is specifically provided, or
  2) _any_ panel from the mod

## To Reproduce

1. Activate `@pixies/gptzero/detect-ai-writing-with-gptzero`
2. In the context menu mod component, in the show sidebar brick, disable `Force Panel`
3. On a page, highlight some text
4. Click the :robot_face: emoji in the text selection menu

## Actual behavior

The sidepanel opens (possibly requires a user gesture) to the Mod Launcher

## Expected behavior

The sidepanel opens to the `AI Detector Side Panel`

## Additional context

- https://pixiebrix.slack.com/archives/C0436P48QHY/p1722971574358689

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Closes #8976 

## Demo

https://www.loom.com/share/4a4cf4cfbb7a440e981e443c313b051a

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/contentScript/sidebarController.tsx [hunk 1]
```diff
@@ -227,17 +227,14 @@ export async function toggleSidebar(): Promise<void> {
  * @param activateOptions options controlling the visible panel in the sidebar
  */
 export async function updateSidebar(
-  activateOptions: ActivatePanelOptions = {},
+  activateOptions: ActivatePanelOptions,
```
Comment (grahamlangford) [inline src/contentScript/sidebarController.tsx:230]: Unnecessary, activateOptions should always be defined

File: src/contentScript/sidebarController.tsx [hunk 2]
```diff
@@ -227,17 +227,14 @@ export async function toggleSidebar(): Promise<void> {
...
   await pingSidebar();
 
   if (!isEmpty(activateOptions)) {
     // The sidebarSlice handles the race condition with the panels loading by keeping track of the latest pending
     // activatePanel request.
-    await sidebarInThisTab.activatePanel(getTimedSequence(), {
-      ...activateOptions,
-      force: activateOptions.force,
-    });
+    await sidebarInThisTab.activatePanel(getTimedSequence(), activateOptions);
```
Comment (grahamlangford) [inline src/contentScript/sidebarController.tsx:237]: Unnecessary spread

File: src/sidebar/protocol.tsx [hunk 3]
```diff
@@ -165,21 +165,21 @@ export async function renderPanels(
 
 export async function activatePanel(
   sequence: TimedSequence,
-  options: ActivatePanelOptions,
+  activateOptions: ActivatePanelOptions,
```
Comment (grahamlangford) [inline src/sidebar/protocol.tsx:168]: Just renaming for consistency

File: src/store/sidebar/sidebarSlice.ts [hunk 4]
```diff
@@ -271,20 +271,21 @@ const sidebarSlice = createSlice({
...
       state.pendingActivePanel = null;
-      const hasActive = state.forms.length > 0 || state.panels.length > 0;
 
-      if (hasActive && !payload.force) {
+      const visiblePanelCount = getVisiblePanelCount(state);
+      const isModLauncherOnlyTabVisible =
+        visiblePanelCount === 1 &&
+        !state.closedTabs[eventKeyForEntry(MOD_LAUNCHER)];
+
+      if (!payload.force && !isModLauncherOnlyTabVisible) {
```
Comment (grahamlangford) [inline src/store/sidebar/sidebarSlice.ts:280]: This is the fix

**Comment by fungairino** [comment]
Does any documentation need to be updated (or description around the force field)?

**Review by fungairino — APPROVED** [review]

**Comment by twschiller** [comment]
> Does any documentation need to be updated (or description around the force field)?

I don't think so? This PR is fixing behavior that was buggy w.r.t. expectations: https://www.notion.so/pixiebrix/8976-show-sidebar-specification-54cb74ac3170407da2bdaa861fc62ea0?pvs=4

File: src/bricks/effects/sidebarEffects.ts [hunk 5]
```diff
@@ -63,7 +63,7 @@ export class ShowSidebar extends EffectABC {
   async effect(
     {
       panelHeading,
-      forcePanel = false,
+      forcePanel = true,
```
Comment (twschiller) [inline src/bricks/effects/sidebarEffects.ts:66]: @grahamlangford This is a backward incompatible change

**Review by twschiller — CHANGES_REQUESTED** [review]
Flagging backward-incompatible change for existing bricks with forcePanel excluded: https://github.com/pixiebrix/pixiebrix-extension/pull/8987/files#r1711602209

I'm open to making the change (as I doubt it will have a big impact and is probably what people already expect, but flagging to at least have the discussion)

File: src/bricks/effects/sidebarEffects.ts (hunk 5 above)
Comment (grahamlangford) [reply src/bricks/effects/sidebarEffects.ts:66]: You're right. I should have caught that. Got a little trigger happy with the switch. I'll revert and update getExampleBrickConfig

**Review by twschiller — APPROVED** [review]
Consider setting example brick config as necessary

**Pull request merged**